### ISSUE TITLE
How easy is it to change the way months are selected?

### ISSUE THREAD

**Issue by thejollygeo** [Blue-Ventures-Conservation/GEM#33]
I know we've spoken about this a couple of times before, including back in Nov 2023, but can't quite remember what we concluded...

See comments which explain the requirement [here](https://docs.google.com/document/d/1bkchkn9iTHtLvflMTBMfnRlNvYIGGho7AxrAGStRkYA/edit?disco=AAAA_8vGWLc)...

A third reason to do this is if a project start or end date is the very beginning (1 Jan) or end (31 Dec) of the year (e.g. 2023).  Let's use the example 31 Dec 2023 as was the case with our ICF mapping efforts.  In this case, the user may want to use a 4-month window comprising 2 months either side: 1 Nov 2023 to 28 Feb 2024.  The way GEM is currently set up, this isn't possible - if the user selects months 11 to 2 for years 2023 and 2024, the unwanted months Jan+Feb 2023 and Nov+Dec 2024 would be included.

So - would implementing this involve significant changes to the code / the way GEM works, or is this something that is relatively easy to implement?  If significant, maybe we could do without, but long-term, versus a user being able to select specific months, the current design does seem a bit restrictive.  If possible, would be great to have this implemented for the GEM v2 release?

**Comment by zibnix** [comment]
It's not a massive change, and I am happy to work on it in the v2 branch!

**Comment by thejollygeo** [comment]
Excellent news!

**Comment by zibnix** [comment]
This is done on v2.

**Comment by zibnix** [comment]
Yes, this hasn't been an option under any of the methods of selecting months we've discussed. In order to accomplish this it might be better to have another option of just entering specific date ranges like 11/2019 - 02/2020.

To stay with the way things work now, selecting months/years separately from each other, you'd have to ask the user about which months matter in each of the years in the range of years, which could get a bit confusing I would imagine.

This is also a break from targeting the dry season, which would typically be the same range of months every year for the historical or contemporary period. Indonesia's dry season is April to October, so it looks like you are targeting the wet season as well.

If you think this is necessary, I can add the option for month/year ranges that would just override the existing method when those settings are used. How does that sound?

**Comment by thejollygeo** [comment]
OK I need to give this real thought, so we remain committed to simplification! Thanks for presenting the options.

One thing that does come to mind is that there are regions where the rainy season is actually quite dry and cloud-free, e.g. where I am in Thailand is currently supposedly a rainy season but it barely rains except in the night, or rains heavy for a couple of hrs then blue skies rest of the day.  So "wet" seasons can still be relatively cloud-free, making the selection of wet season months plausible in some places.
